AdobeIms/Test/Unit/Model/ConfigTest.php

```diff
@@ -36,28 +30,129 @@ class ConfigTest extends TestCase
     private $scopeConfigMock;
 
     /**
-     * Prepare test objects.
+     * @var UrlInterface|MockObject
+     */
+    private $urlMock;
+
+    /**
+     * Set up test mock objects
      */
     protected function setUp(): void
     {
-        $this->objectManager = new ObjectManager($this);
         $this->scopeConfigMock = $this->createMock(ScopeConfigInterface::class);
-        $this->config = $this->objectManager->getObject(
-            Config::class,
-            [
-                'scopeConfig' => $this->scopeConfigMock
-            ]
-        );
+        $this->urlMock = $this->createMock(UrlInterface::class);
+
+        $this->config = new Config($this->scopeConfigMock, $this->urlMock);
     }
 
     /**
-     * Get API key test.
+     * Test for \Magento\AdobeIms\Model\Config::getApiKey
      */
     public function testGetApiKey(): void
     {
         $this->scopeConfigMock->expects($this->once())
             ->method('getValue')
-            ->with(self::CONFIG_XML_PATH_API_KEY);
+            ->with(Config::XML_PATH_API_KEY);
+
         $this->config->getApiKey();
     }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getPrivateKey
+     */
+    public function testGetPrivateKey(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_PRIVATE_KEY);
+
+        $this->config->getPrivateKey();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getTokenUrl
+     */
+    public function testGetTokenUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_TOKEN_URL)
+            ->willReturn('URL');
+
+        $this->config->getTokenUrl();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getAuthUrl
+     */
+    public function testGetAuthUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->exactly(3))
+            ->method('getValue')
+            ->withConsecutive(
+                [Config::XML_PATH_API_KEY],
+                [Custom::XML_PATH_GENERAL_LOCALE_CODE],
+                [Config::XML_PATH_AUTH_URL_PATTERN]
+            )
+            ->willReturn('URL');
+
+        $this->urlMock->expects($this->once())->method('getUrl')->willReturn('URL');
+
+        $this->config->getAuthUrl();
```

Comment:
Please check that correct value is returned instead of asserting the number of internal calls